inference/fold_detection.py

```diff
@@ -0,0 +1,73 @@
+import sys
+import torch
+import json
+from args import get_argparser, parse_args, get_aligner, get_bbox, get_provenance
+from os.path import join
+from cloudmanager import CloudManager
+from time import time
+if __name__ == '__main__':
+  parser = get_argparser()
+  parser.add_argument('--model_path', type=str,
+    help='relative path to the ModelArchive to use for computing fields')
+  parser.add_argument('--src_path', type=str)
+  parser.add_argument('--src_mask_path', type=str, default='',
+    help='CloudVolume path of mask to use with src images; default None')
+  parser.add_argument('--src_mask_mip', type=int, default=8,
+    help='MIP of source mask')
+  parser.add_argument('--src_mask_val', type=int, default=1,
+    help='Value of of mask that indicates DO NOT mask')
+  parser.add_argument('--dst_path', type=str)
+  parser.add_argument('--mip', type=int)
+  parser.add_argument('--bbox_start', nargs=3, type=int,
+    help='bbox origin, 3-element int list')
+  parser.add_argument('--bbox_stop', nargs=3, type=int,
+    help='bbox origin+shape, 3-element int list')
+  parser.add_argument('--bbox_mip', type=int, default=0,
+    help='MIP level at which bbox_start & bbox_stop are specified')
+  parser.add_argument('--max_mip', type=int, default=9)
+  parser.add_argument('--max_displacement', 
+    help='the size of the largest displacement expected; should be 2^high_mip', 
+    type=int, default=2048)
+  parser.add_argument('--block_size', type=int, default=10)
+  args = parse_args(parser)
+  # Only compute matches to previous sections
+  args.serial_operation = True
+  a = get_aligner(args)
+  bbox = get_bbox(args)
+  provenance = get_provenance(args)
+  
+  # Simplify var names
+  mip = args.mip
+  max_mip = args.max_mip
+  pad = args.max_displacement
+  chunk_size = (256, 256)
+
+  # Compile ranges
+  full_range = range(args.bbox_start[2], args.bbox_stop[2])
+  # Create CloudVolume Manager
+  cm = CloudManager(args.src_path, max_mip, pad, provenance, batch_size=1,
+                    size_chunk=256, batch_mip=mip)
+
+  # Create src CloudVolumes
+  src = cm.create(args.src_path, data_type='uint8', num_channels=1,
+                     fill_missing=True, overwrite=False)
+
+  # Create dst CloudVolumes
+  dst = cm.create(join(args.dst_path, 'image'),
+                  data_type='uint8', num_channels=1, fill_missing=True,
+                  overwrite=True)
+  
+  chunks = a.break_into_chunks(bbox, chunk_size,
+                                 cm.dst_voxel_offsets[mip], mip=mip,
+                                 max_mip=cm.num_scales)
+
+  ###########################
+  # Serial alignment script #
+  ###########################
+  
+  n_chunks = len(chunks) 
+  for z in full_range:
+      print("fold detection for z={}".format(z))
+      a.fold_detection(cm, args.model_path, src, dst, z, mip, bbox,
```

Comment:
You'll need to update this with the `batch` then `run` convention. See the latest `serial_block_alignment.py` script.